### PR TITLE
MM-33731: Fix for Incidents can be created without a name

### PR DESCRIPTION
**Pull request by cpoile**
#### Summary
- Do the check in the createIncident function
- Add a test

#### Ticket Link
- https://mattermost.atlassian.net/browse/MM-33731

### CODE REVIEW

File: server/api/incidents_test.go [hunk 1]
```diff
@@ -649,6 +649,26 @@ func TestIncidents(t *testing.T) {
...
+		}
+
+		pluginAPI.On("GetChannel", mock.Anything).Return(&model.Channel{}, nil)
+		pluginAPI.On("HasPermissionToTeam", "testUserID", "testTeamID", model.PERMISSION_CREATE_PUBLIC_CHANNEL).Return(true)
+		pluginAPI.On("HasPermissionToTeam", "testUserID", "testTeamID", model.PERMISSION_VIEW_TEAM).Return(true)
+
+		resultIncident, err := c.Incidents.Create(context.TODO(), icClient.IncidentCreateOptions{
+			TeamID:          testIncident.TeamID,
+			CommanderUserID: testIncident.CommanderUserID,
+		})
```
Comment (agarciamontoro) [inline server/api/incidents_test.go:667]: Nit: can we be explicit with an empty string here?
```suggestion
		resultIncident, err := c.Incidents.Create(context.TODO(), icClient.IncidentCreateOptions{
			Name:            "",
			TeamID:          testIncident.TeamID,
			CommanderUserID: testIncident.CommanderUserID,
		})
```

**Review by agarciamontoro — APPROVED** [review]
Good stuff! I left a nit suggestion, but nothing blocking. Thank you, Christopher!

**Review by crspeller — APPROVED** [review]

**Comment by cpoile** [comment]
@prapti Ready for you, thanks!

**Review by prapti — APPROVED** [review]
Original bug is fixed.
Approving!

**Pull request merged**